### ISSUE TITLE
Is it possible to update p with loss of another parameter p_?

### ISSUE THREAD

**Issue by Waterkin** [deepmind/optax#340]
for example: 

`grads = grad(loss)(p2, x, y)`
`inner_sgd_fn = lambda g, state: (state - alpha*g)`
`p = tree_multimap(inner_sgd_fn, grads, p2)`

or

`grads = jax.grad(loss)(p2, x, y) `
`updates, state = opt.update(grads, state) `
`p = optax.apply_updates(p, updates)`

**Comment by n2cholas** [comment]
Yes, nothing in the API would prevent you from doing so. The gradient computation from the loss is completely separate from applying the updates in Optax (as you showed in your code snippet). Are there any issues you encountered when trying to do this?

**Comment by mtthss** [comment]
As long as `p` and `p2` that seems fine.

**Comment by mkunesch** [comment]
Closing this as I think the question has been answered.
